### PR TITLE
implement /api/v1/utxo?block=<height>

### PR DESCRIPTION
**Pull request by cleverca22**
this is currently based on the branch from https://github.com/input-output-hk/cardano-explorer/pull/45, if that gets merged, then this pr will only show 1 new endpoint in the diff

the sql is also currently wrong, and returns different data as more tx's get inserted, so the filter by block height isnt working

### CODE REVIEW

File: cardano-explorer/src/Explorer/Web/LegacyApi.hs [hunk 1]
```diff
@@ -16,6 +16,7 @@ module Explorer.Web.LegacyApi
        ) where
 
 import           Data.Word                (Word16, Word64)
+import           Data.Int (Int64)
```
Comment (erikd) [inline cardano-explorer/src/Explorer/Web/LegacyApi.hs]: I am pretty sure we are never going to run this on a platform where `Int` is not 64 bits, so I would thing `Int` is just fine for this if you needed a signed value.

File: cardano-explorer/src/Explorer/Web/Query.hs [hunk 2]
```diff
@@ -127,3 +128,47 @@ queryTx hash = do
...
+data TxWithInputsOutputs = TxWithInputsOutputs
+  { txwTx :: Tx
+  , txwInputs :: [(Text, Word64)]
+  , txwOutputs :: [TxOut]
+  }
+
+queryBlockTxs :: MonadIO m => ByteString -> Int64 -> Int64 -> ReaderT SqlBackend m ([ TxWithInputsOutputs ], Word64)
+queryBlockTxs blkHash limitNum offsetNum = do
+  let
+    blockid = subList_select . from $ \blk -> do
```
Comment (erikd) [inline cardano-explorer/src/Explorer/Web/Query.hs:190]: Do you think using `where` causes might make this clearer?

`where` clauses allow you to put the meat of the function close to the type signature and helper functions in the `where.

File: cardano-explorer/src/Explorer/Web/Server.hs [hunk 3]
```diff
@@ -179,90 +192,116 @@ getBlocksPagesTotal backend mPageSize = do
     (_, True) -> pure $ Left $ Internal "Page size must be greater than 1 if you want to display blocks."
     _ -> pure $ Right $ roundToBlockPage blocksTotal
 
+hexToBytestring :: T.Text -> ExceptT ExplorerError Handler BS.ByteString
+hexToBytestring text = do
+  case B16.decode (SB8.pack (T.unpack text)) of
+    (blob, "") -> pure blob
+    _ -> throwE $ Internal "cant parse hex"
```
Comment (erikd) [inline cardano-explorer/src/Explorer/Web/Server.hs]: Might be useful to add the part that failed to parse to the error message.

File: cardano-explorer/src/Explorer/Web/Server.hs [hunk 4]
```diff
@@ -179,90 +192,116 @@ getBlocksPagesTotal backend mPageSize = do
...
+
+bytestringToText :: BS.ByteString -> T.Text
+bytestringToText = T.pack . SB8.unpack . B16.encode
```
Comment (erikd) [inline cardano-explorer/src/Explorer/Web/Server.hs]: `T.pack . SB8.unpack` can be replaced with `T.decodeUtf8` (in `Data.Text.Encoding`).

**Review by erikd — APPROVED** [review]

File: cardano-explorer/src/Explorer/Web/LegacyApi.hs (hunk 1 above)
Comment (cleverca22) [reply cardano-explorer/src/Explorer/Web/LegacyApi.hs]: that was set by the type of:
```
limit :: Int64 -> SqlQuery ()
offset :: Int64 -> SqlQuery ()
```
i didnt see a need to change the type and then add `fromIntegral`s everywhere to restore the old type

Comment (erikd) [reply cardano-explorer/src/Explorer/Web/LegacyApi.hs]: Ok, that makes sense then.

**Pull request merged**